BangazonAPI/BangazonAPI/Controllers/ProductController.cs

```diff
@@ -29,37 +31,248 @@ public SqlConnection Connection
+        public async Task<IActionResult> Put([FromRoute] int id, [FromBody] Product Product)
         {
+            try
+            {
+                using (SqlConnection conn = Connection)
+                {
+                    conn.Open();
+                    using (SqlCommand cmd = conn.CreateCommand())
+                    {
+                        cmd.CommandText = @"UPDATE Product
+                                            SET ProductTypeId=@ProductTypeId, 
+                                            CustomerId=@CustomerId, 
+                                            Price= @Price, 
+                                            [Title]=@Title,
+                                            [Description]=@Description,
+                                            Quantity=@Quantity,
+                                            Archived=0
+                                            WHERE Id = @id";
+                        cmd.Parameters.Add(new SqlParameter("@ProductTypeId", Product.ProductTypeId));
+                        cmd.Parameters.Add(new SqlParameter("@CustomerId", Product.CustomerId));
+                        cmd.Parameters.Add(new SqlParameter("@Price", Product.Price));
+                        cmd.Parameters.Add(new SqlParameter("@Title", Product.Title));
+                        cmd.Parameters.Add(new SqlParameter("@Description", Product.Description));
+                        cmd.Parameters.Add(new SqlParameter("@Quantity", Product.Quantity));
+                        cmd.Parameters.Add(new SqlParameter("@id", id));
+
+                        int rowsAffected = cmd.ExecuteNonQuery();
+                        if (rowsAffected > 0)
+                        {
+                            return new StatusCodeResult(StatusCodes.Status204NoContent);
+                        }
+                        throw new Exception("No rows affected");
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                if (!ProductExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
         }
 
-        // DELETE: api/Product/5
         [HttpDelete("{id}")]
-        public void Delete(int id)
+        public async Task<IActionResult> Delete([FromRoute] int id, bool delete)
         {
+            try
+            {
+                using (SqlConnection conn = Connection)
+                {
+                    conn.Open();
+                    using (SqlCommand cmd = conn.CreateCommand())
+                    {
+                        if (delete == true)
+                        {
+                            cmd.CommandText = @"DELETE FROM Product WHERE Id = @id";
+                        }
+                        else
+                        {
+                            cmd.CommandText = @"UPDATE PaymentType SET Archived=1 WHERE Id =@id";
```

Comment:
Don't delete PaymentType

Reply:
Should be fixed now Kim

Reply:
Now code shall update soft delete change on get method.

Reply:
Now code shall update soft delete change on get method.